### ISSUE TITLE
Rebuild menu on app focus gain because Unity is a jerk

### ISSUE THREAD

**Issue by LiamDawe** [itchio/itch#124]
Both Infinitrap and Reckpunk seem to install fine, but launching them from that app doesn't work.

They work if I manually load up the directory and launch the files directly.

The Itch app doesn't seem to output anything if ran in terminal, so I can't send you any logs :(

**Comment by fasterthanlime** [comment]
> The Itch app doesn't seem to output anything if ran in terminal

I know you have a Gaming on Linux avatar, but... are you running it from MSYS? if so you need to export `OS=cygwin`.

If you're runninng it on OSX, you need to run the binary directly, not use `open` on the .app

If you're running it on Linux, are you using the official .deb? It definitely logs things for me.

The download log you posted in your other issue is also an install/run log and is accessible by alt-clicking on the game's thumbnails.

**Comment by LiamDawe** [comment]
Sorry, should have stated this is on Ubuntu Linux. 

Yes downloaded using official deb (the USC also said the .deb was "bad quality"?).

Alt-click on thumbnail does nothing for me.

**Comment by fasterthanlime** [comment]
> the USC also said the .deb was "bad quality"?

working on it, cf. #93 

> Alt-click on thumbnail does nothing for me.

can you confirm you're running 0.9.3 ? ('Help' menu)

**Comment by LiamDawe** [comment]
Hmm, seems to be no help menu for me?

**Comment by fasterthanlime** [comment]
@NaughtySquid just checked on my ubuntu box, it seems Unity hides menus by default now? Anyway if you hover over the menu bar it should appear.

**Comment by LiamDawe** [comment]
Unity by default shows a menu in the unity bar, not on the application. Itch app doesn't seem to have one at all, even after switching it to be a local menu in the setting.

**Comment by fasterthanlime** [comment]
Which version of Ubuntu are you running? I can't reproduce either lack of log output or lack of menu bar

**Comment by LiamDawe** [comment]
Ubuntu 15.10, 64bit.

**Comment by fasterthanlime** [comment]
> Ubuntu 15.10, 64bit.

Actually I have a VM of that, let me try again

**Comment by fasterthanlime** [comment]
@NaughtySquid here's an ubuntu 15.10 amd64 screenshot with itch 0.9.3  - hovering on the menu bar makes it appear:

![ubuntu-menu](https://cloud.githubusercontent.com/assets/7998310/11813435/cf42e230-a341-11e5-9ed4-06558f749746.png)

**Comment by fasterthanlime** [comment]
@NaughtySquid I think I know what's happening. When the window is hidden, the menu is destroyed on Unity.

It needs to be rebuilt when the app is activated

**Comment by LiamDawe** [comment]
Yep, can confirm on first load it's there, bringing it back up from the tray causes the menu to vanish.

**Comment by fasterthanlime** [comment]
Alright, then I have a pretty good idea how to fix it. Too late for 0.9.4 but I'll keep it in mind for the next one

**Comment by LiamDawe** [comment]
Awesome, thanks.
